apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx

```diff
@@ -108,12 +119,81 @@ type WebstudioComponentDevProps = {
   components: Components;
 };
 
-export const WebstudioComponentDev = ({
-  instance,
-  instanceSelector,
-  children,
-  components,
-}: WebstudioComponentDevProps) => {
+/**
+ * Radix's VisuallyHiddenPrimitive.Root https://github.com/radix-ui/primitives/blob/main/packages/react/visually-hidden/src/VisuallyHidden.tsx
+ * component makes content from hidden elements accessible to screen readers.
+ * Using the same Content children, however, leads to duplicated React elements, breaking our 'isSelected' logic.
+ * To prevent this, we check if an instance is a descendant of VisuallyHiddenPrimitive.Root, and if so,
+ * we avoid rendering it.
+ */
+const useIsScreenReaderDescendant = (ref: RefObject<HTMLElement>) => {
+  const [isScreenReaderDescendant, setIsScreenReaderDescendant] =
+    useState(false);
+
+  useEffect(() => {
+    if (ref.current !== null) {
+      for (
+        let element: HTMLElement | null = ref.current;
+        element !== null;
+        element = element.parentElement
+      ) {
+        if (
+          element.style.overflow === "hidden" &&
+          element.style.clip === "rect(0px, 0px, 0px, 0px)" &&
+          element.style.position === "absolute" &&
+          element.style.width === "1px" &&
+          element.style.height === "1px"
+        ) {
+          setIsScreenReaderDescendant(true);
+          return;
+        }
+      }
+    }
+  }, [ref]);
```

Comment:
I don't quite get it but it shouldn't need a ref as a dependency, no? or is it just for linter, if only for linter, then I would rather make the disable eslint comment because it makes it clear